pkg/api/grpc/grpc.go

```diff
@@ -505,23 +508,32 @@ func (a *api) InvokeBinding(ctx context.Context, in *runtimev1pb.InvokeBindingRe
 	sc := span.SpanContext()
 	tp := diag.SpanContextToW3CString(sc)
 	if span != nil {
-		if _, ok := req.Metadata[diag.TraceparentHeader]; !ok {
-			req.Metadata[diag.TraceparentHeader] = tp
+		if _, ok := req.Metadata[diagConsts.TraceparentHeader]; !ok {
+			req.Metadata[diagConsts.TraceparentHeader] = tp
 		}
-		if _, ok := req.Metadata[diag.TracestateHeader]; !ok {
+		if _, ok := req.Metadata[diagConsts.TracestateHeader]; !ok {
 			if sc.TraceState().Len() > 0 {
-				req.Metadata[diag.TracestateHeader] = diag.TraceStateToW3CString(sc)
+				req.Metadata[diagConsts.TracestateHeader] = diag.TraceStateToW3CString(sc)
 			}
 		}
 	}
 
 	// Allow for distributed tracing by passing context metadata.
 	if incomingMD, ok := metadata.FromIncomingContext(ctx); ok {
+		if baggageValues := incomingMD[diagConsts.BaggageHeader]; len(baggageValues) > 0 {
+			baggageString := strings.Join(baggageValues, ",")
+			baggage, err := otelbaggage.Parse(baggageString)
+			if err == nil {
+				ctx = otelbaggage.ContextWithBaggage(ctx, baggage)
+				req.Metadata[diagConsts.BaggageHeader] = baggageString
+			}
```

Comment:
Why are are ignoring the error here, rather than returning?

Reply:
If the baggage parsing succeeds, then we add the baggage. Otherwise, we simply skip adding the baggage - this way we still proceed with the request rather than failing it entirely due to baggage parsing. I can fail the request if that is preferred, but figured simply not propagating it along made sense. Open to additional opinions here :)

Reply:
We should always reject requests which have invalid inputs, else we end up with magic APIs which will silently ignore input. This leads to frustrated/confused users and possible catastrophic downstream results. Please reject the request on malformed baggage headers.